src/signTx.c

```diff
@@ -349,26 +382,56 @@ static void signTx_handleInit_ui_runStep()
 
 	UI_STEP_BEGIN(ctx->ui_step, this_fn);
 
-	UI_STEP(HANDLE_INIT_STEP_DISPLAY_DETAILS) {
-		char* header =
-		        (ctx->commonTxData.txSigningMode == SIGN_TX_SIGNINGMODE_MULTISIG_TX) ?
-		        "Multisig transaction" :
-		        "New transaction";
+	UI_STEP(HANDLE_INIT_STEP_PROMPT_SIGNINGMODE) {
+		char bodyTxt[SIZEOF("Pool operator") + 1 + SIZEOF("transaction?") + 1];
+		const uint32_t signingModeStrIndex = ctx->commonTxData.txSigningMode - SIGN_TX_SIGNINGMODE_ORDINARY_TX;
+		ASSERT(signingModeStrIndex < ARRAY_LEN(uiSigningModeName));
```

Comment:
```suggestion
		ASSERT(signingModeStrIndex >= 0);
		ASSERT(signingModeStrIndex < ARRAY_LEN(uiSigningModeName));
```

Reply:
It's an unsigned integer, why do we want to assert it being non-negative?

Reply:
I overlooked that somehow, no need for that assert.